tests/async_engine/test_merge_async_iterators.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+from typing import AsyncIterator, Tuple
+
+import pytest
+
+from vllm.utils import merge_async_iterators
+
+
+@pytest.mark.asyncio
+async def test_merge_async_iterators():
+
+    async def mock_async_iterator(idx: int) -> AsyncIterator[str]:
+        try:
+            while True:
+                yield f"item from iterator {idx}"
+                await asyncio.sleep(0.1)
+        except asyncio.CancelledError:
+            pass
+
+    iterators = [mock_async_iterator(i) for i in range(3)]
+    merged_iterator: AsyncIterator[Tuple[int, str]] = merge_async_iterators(
+        *iterators)
+
+    async def stream_output(generator: AsyncIterator[Tuple[int, str]]):
+        async for idx, output in generator:
+            print(f"idx: {idx}, output: {output}")
+
+    task = asyncio.create_task(stream_output(merged_iterator))
+    await asyncio.sleep(0.5)
+    task.cancel()
+    with pytest.raises(asyncio.CancelledError):
+        await task
+
+    for iterator in iterators:
+        try:
+            await anext(iterator)
```

Comment:
```suggestion
            await asyncio.wait_for(anext(iterator), 1)
```

since I think without your fix the test would hang indefinitely?

Reply:
Actually, the unfixed code raises the following error: `RuntimeError: anext(): asynchronous generator is already running`. However, your suggestion helps prevent the risk of the code hanging indefinitely during test, so I've decided to adopt it. Thanks a lot for your advice!